### PR TITLE
setting up a wallet made super simple

### PR DESCRIPTION
**Pull request by hschoenburg**
very simple guide

### CODE REVIEW

**Comment by hschoenburg** [comment]
further topics to cover

- basic auction functions
- cli versus rpc

-links to documentation

-deriving a new sending address

--viewing account balance

File: src/guides/wallet.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+### The Basics
+If you're a seasoned bitcoiner, you can probably skim this section or skip straight ahead to the <a href='#examples'> Examples </a>section. But if you're relatively new or just want a refresher, this section will help you understand how wallets actually work.
+
+### Wallets
+In the most basic sense, a handshake wallet is data that enables you to receive and spend HNS, place bids on names and update resource records for your names. `hsd` implements the latest specifications for structuring wallets that are easy to backup, easy to restore, and that work just as well for a novice making their first transactions as for a business with millions of users.
+
+### Keys to the Game
+If you want to transact with HNS, you'll need keys. Each handshake address is associated with a particular key, and wallets are made up of many different keys. Keys consist of both a private key and a public key. The private key is required for spending and is extremely sensitive information, while a public key can be used to receive HNS  and monitor a particular address. If you want to learn more about how this works, read up on Public-Key Cryptography.
```
Comment (boymanjor) [inline src/guides/wallet.md]: nit: The line, "while a public key can be used to receive HNS  and monitor" has a double space here: "HNS  and"

Comment (boymanjor) [inline src/guides/wallet.md]: nit: The line, "while a public key can be used to receive HNS and monitor" has a double space here: "HNS and"

File: src/guides/wallet.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+### HD vs. Non-HD, you mean like TVs?
+You may have seen references to "HD" wallets and wondered what that means. HD in this context does not mean "high definition," as I assumed it did at first, but rather "hierarchical deterministic." An HD wallet takes a hierarchy of keys in order and makes it so any key in that sequence can be determined by the one before it. This means that if you can produce the first key in the hierarchy, you can then generate a practically unlimited number of subsequent keys. The specification for HD wallets as implemented in `hsd` is defined by [Bitcoin Improvement Proposal (BIP) 32](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki).
```
Comment (boymanjor) [inline src/guides/wallet.md]: This needs a better explanation. Specifically, the description of the keys does not illustrate that there is a tree of keys. Each level specifies a different piece of information. The first level specifies that the wallet is adhering to bip44. The second specifies what coin, i.e. 0' for Bitcoin, etc. Perhaps this is too much information but what is here now is a bit too vague and also technically incorrect.

File: src/guides/wallet.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+### Watch Only Wallets
+Speaking of not touching private keys, hsd gives you the option to create wallets that are "watch only." Watch only wallets don't contain any private keys, which means they can't be used to spend the HNS they receive. However, they work perfectly fine for creating addresses, receiving HNS, and detecting incoming transactions. Using watch only wallets where appropriate reduces the risk of your keys and HNS being stolen and is good security practice.
+
+Accounts always inherit the watch only behavior of their parent wallet. In other words, a watch only wallet will have exclusively watch only accounts while a regular wallet will have only regular accounts. Accordingly, you can't import private keys into a watch only wallet or public keys into regular wallets. If you try to mix and match watch only wallets and keys with hsd, you're gonna have a bad time.
+
+
```
Comment (boymanjor) [inline src/guides/wallet.md]: nit: double newline.

File: src/guides/wallet.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+### API Authentication
+`hsd` runs as a server and allow you to interact with your wallets via a REST API. It also allows you protect wallets from unauthenticated requests by running the server with the wallet-auth option. Each wallet you create has a token value that must be passed in with each request. Tokens, like accounts and keys, can also be deterministically generated using your HD seed. This means you can change the token on a wallet as often as you'd like.
+
+### Recovery
+By using the HD standards mentioned above, hsd allows one to easily restore or transfer their entire wallet to different wallet implementations. By providing just the mnemonic, one can fully recover their wallet to a fresh instance of `hsd` or any other software that properly implements BIP33, BIP39, and BIP44, like the Trezor hardware wallet.
```
Comment (boymanjor) [inline src/guides/wallet.md]: I wouldn't mention a particular wallet. Someone would still need to write software to support this wallet.

File: src/guides/wallet.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
+## Seting up a Handshake Wallet 
+
+
```
Comment (boymanjor) [inline src/guides/wallet.md]: nit: double newline.

File: src/guides/wallet.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+### API Authentication
+`hsd` runs as a server and allow you to interact with your wallets via a REST API. It also allows you protect wallets from unauthenticated requests by running the server with the wallet-auth option. Each wallet you create has a token value that must be passed in with each request. Tokens, like accounts and keys, can also be deterministically generated using your HD seed. This means you can change the token on a wallet as often as you'd like.
+
+### Recovery
+By using the HD standards mentioned above, hsd allows one to easily restore or transfer their entire wallet to different wallet implementations. By providing just the mnemonic, one can fully recover their wallet to a fresh instance of `hsd` or any other software that properly implements BIP33, BIP39, and BIP44, like the Trezor hardware wallet.
+
+*Above borrowed from [Wallets and Accounts and Keys, Oh My!](https://bcoin.io/guides/wallets.html) by Daniel McNally*
```
Comment (boymanjor) [inline src/guides/wallet.md]: I would mention this at the very top of this guide, and use stronger language, e.g. "this guide is heavily influenced by such and such....much of it is copied verbatim with name changes from Bitcoin to Handshake." Or something like this...

File: src/guides/wallet.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+* `hsd-cli` - hsd node REST API
+&nbsp;\
+
+
+For more info on node commands unrelated to the wallet. [See here](https://handshake-org.github.io/api-docs))
+
+Below you will find short intros to both wallet tools `hsw-cli` and `hsw-rpc` Use them for wallet creation, sending coins, generating addresses, bidding on names, updating resource records for a name, transfering names, etc.. 
+&nbsp;\
+
+
```
Comment (boymanjor) [inline src/guides/wallet.md]: nit: double newline.

File: src/guides/wallet.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+### Set the network option
+
+You have four choices to chose when setting your `hsd` node's network. The `hs-client` CLI tools also require this option be set for them to know the port with which to connect to your `hsd` node.
```
Comment (boymanjor) [inline src/guides/wallet.md]: "choices to chose"

File: src/guides/wallet.md [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+
+* `regtest` Best for local development work.
+* `testnet` Closest simulation to mainnet. Supported by a network of nodes run by other devs.
+* `mainnet` Live production.
```
Comment (boymanjor) [inline src/guides/wallet.md]: Above we mention four options but only list 3.

File: src/guides/wallet.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+
+You can set the network for your hsd node with the `--network=` option. Or with `$ export HSD_NETWORK=`
```
Comment (boymanjor) [inline src/guides/wallet.md]: I would mention setting an environment variable as opposed to just saying to use `export`.

File: src/guides/wallet.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+&nbsp;\
+See that `token`? With it we can do things like query our wallet balance
+``` bash
+$ hsw-cli balance --id=mynewwallet --token=aef7603ff78c32e267fad434246ce6447d420b81e4798d99cf06c80b57c40765
+```
+
+&nbsp;\
+Generate a new receiving address for an account
+``` bash
+$ hsw-cli --id=$name-of-wallet --accout=$name-of-account address
```
Comment (boymanjor) [inline src/guides/wallet.md]: --accout

File: src/guides/wallet.md [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+
+&nbsp;\
+Send HNS coins to an address
+``` bash
+$ hsw-cli send 
+```
+&nbsp;\
+
+### hsw-rpc
+Handshake Name specific functionality
```
[the remark this answers is not in the document]
Comment (boymanjor) [reply src/guides/wallet.md]: You did not add the additional commands for sending bids and registering names.

File: src/guides/wallet.md (hunk 11 above)
Comment (boymanjor) [reply src/guides/wallet.md]: Also, a common notation is to use brackets for required CLI options, e.g.
```
$ hsw-cli --id=<wallet name> --account=<account name> address
```

File: src/guides/wallet.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+&nbsp;\
+Generate a new receiving address for an account
+``` bash
+$ hsw-cli --id=$name-of-wallet --accout=$name-of-account address
+```
+
+&nbsp;\
+Get the HNS balance of your wallet
+``` bash
+$ hsw-cli --id=balance
```
Comment (boymanjor) [inline src/guides/wallet.md]: This isn't correct.

File: src/guides/wallet.md [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+&nbsp;\
+List all wallets in your walletdb.
+``` bash
+$ hsw-cli wallets
+```
+
+&nbsp;\
+Send HNS coins to an address
+``` bash
+$ hsw-cli send
```
Comment (boymanjor) [inline src/guides/wallet.md]: This is not correct. There are other arguments that need to be specified to send.

File: src/guides/wallet.md (hunk 2 above)
Comment (boymanjor) [reply src/guides/wallet.md]: Proper handling of this description also sets the table for explaining accounts, which are specified in the third level.

File: src/guides/wallet.md (hunk 4 above)
Comment (boymanjor) [reply src/guides/wallet.md]: Perhaps say something like, "that properly implements, blah, blah blah, and also has support for Handshake."

File: src/guides/wallet.md [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+*Watch carefully how values are entered in the examples, all examples send the same amount when executed*
+
+``` bash
+$ hsw-cli send --id=<name-of-wallet> --value=<number of WHOLE HNS> --address=<destination address> --passphrase=<passphrase>
+```
+
+&nbsp;\
+Initiate blockchain rescan for walletdb. Necessary after importing a key or wallet. Wallets will be rolled back to the specified block height if provided.
+``` bash
+$ hsw-cli rescan --height<block-height>
```
Comment (boymanjor) [inline src/guides/wallet.md]: We need a space or equals sign separating `--height` and `<block-height>`.

File: src/guides/wallet.md [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+Below you will find short intros to both wallet tools `hsw-cli` and `hsw-rpc` Use them for wallet creation, sending coins, generating addresses, bidding on names, updating resource records for a name, transfering names, etc.. 
+&nbsp;\
+
+### Set the network option
+
+You have three choices when setting your `hsd` node's network. The `hs-client` CLI tools also require this option be set for them to know the port with which to connect to your `hsd` node.
+
+* `regtest` Best for local development work.
+* `testnet` Closest simulation to mainnet. Supported by a network of nodes run by other devs.
+* `mainnet` Live production.
```
[the remark this answers is not in the document]
Comment (hschoenburg) [reply src/guides/wallet.md]: Did I not resolve this one?

[the remark this answers is not in the document]
Comment (boymanjor) [reply src/guides/wallet.md]: There are four options, not three. simnet...

[the remark this answers is not in the document]
Comment (hschoenburg) [reply src/guides/wallet.md]: oooh ok forgot simnet

[the remark this answers is not in the document]
Comment (hschoenburg) [reply src/guides/wallet.md]: Ok added in simnet

[the remark this answers is not in the document]
Comment (boymanjor) [reply src/guides/wallet.md]: This is great!

**Pull request merged**